### ISSUE TITLE
Error when using `mypy.bogus_type.Bogus`

### ISSUE THREAD

**Issue by adamtheturtle** [microsoft/pyright#9617]
**Describe the bug**

I am implementing a `mypy` plugin and as part of that I use `ctx.default_signature.copy_modified` within a `get_function_signature_hook` hook.

This uses the `Bogus` type, defined at https://github.com/python/mypy/blob/d33cef8396c456d87db16dce3525ebf431f4b57f/mypy/bogus_type.py#L1C1-L27C32:

```python
# Source copied from mypy

from __future__ import annotations

from typing import Any, TypeVar

from mypy_extensions import FlexibleAlias

T = TypeVar("T")

# This won't ever be true at runtime, but we consider it true during
# mypyc compilations.
MYPYC = False
if MYPYC:
    Bogus = FlexibleAlias[T, Any]
else:
    Bogus = FlexibleAlias[T, T]
```

I get an error I don't know how to work around (apart from disabling the error).

**Code or Screenshots**

```
# requirements.txt
mypy==1.14.0
```

```python
# example.py
from mypy.bogus_type import Bogus


def foo(item: Bogus[bool]) -> int:
    return 1
```

Result:

```
> pyright example.py
/Users/adam/Documents/repositories/vws-web-tools/example.py
  /Users/adam/Documents/repositories/vws-web-tools/example.py:5:9 - error: Type of parameter "item" is unknown (reportUnknownParameterType)
  /Users/adam/Documents/repositories/vws-web-tools/example.py:5:15 - error: Variable not allowed in type expression (reportInvalidTypeForm)
2 errors, 0 warnings, 0 informations 
```

* Is this a bug/limitation in pyright?
* Is there a way to use this without ignores?
* Do you have any suggestions for changes to `mypy` which I might be able to contribute to avoid this error?

**VS Code extension or command-line**

pyright 1.1.391 CLI.
Also, pyright in VS Code.

**Comment by erictraut** [comment]
Pyright is working in accordance with the Python typing spec here, so this isn't a bug in Pyright.

A type alias can have only one definition, and it cannot be conditional. You're attempting to define a type alias `Bogus` with multiple (conditional) definitions, so pyright doesn't treat this as a valid type alias.

You can work around this by using the "defineConstant" configuration setting to tell pyright that `MYPYC` should be treated as always false.

Here's what you'd need to add to your pyrightconfig.json file.

```json
"defineConstant": {"MYPYC": false}
```

If you're using pyproject.toml, you'll need to convert this to the relevant toml syntax.

**Comment by adamtheturtle** [comment]
@erictraut Extremely kind of you to give me that solution, thank you!
